### ISSUE TITLE
Language settings confusion

### ISSUE THREAD

**Issue by KrOstir** [streetsidesoftware/vscode-spell-checker#1985]
In the settings there are several versions of the same language, sometimes with different regions. This is OK, but the logic is not clear. E.g., there three different versions of the English language (the code is once with underscore, another time with dash) and two for Slovenian.

![Code_220601_115028](https://user-images.githubusercontent.com/13023733/171379064-aa33d0cb-fa42-41a6-aa47-e8ae744729aa.png)

`en_us`: English
`en-gb`: English - United Kingdom
`en_us`: English - United States
`sl-si`: Slovenian
`sl-si`: Slovenian - Slovenia

I am not sure where this comes from, but it is confusing. What I would expect in the case above is.

`en`: English
`en-gb`: English - United Kingdom
`en-us`: English - United States
`sl`: Slovenian
`sl-si`: Slovenian - Slovenia

**Comment by Jason3S** [comment]
@KrOstir,

Yes, it is confusing. You are seeing the _name_ of the dictionary and not the locale. Maybe I should put the locale after the language name: English<sup>en</sup>, Slovenian<sup>sl</sup>, English - United States<sup>en-us</sup>.

**Comment by KrOstir** [comment]
That might be better, yes.

Just a question, `sl` and `sl-si` are the same dictionary, could just one be listed? The same is true for `en` and `en_us`.

**Comment by Jason3S** [comment]
> That might be better, yes.
> 
> Just a question, `sl` and `sl-si` are the same dictionary, could just one be listed? The same is true for `en` and `en_us`.

It writes different values to `cSpell.language`.

If `cSpell.languageSettings.*.locale` contains a value that matches a value in `cSpell.language`, then the settings are applied. `en` and `en-us` are considered different values.

For example, if your organization wanted to British English to be the default. They could use the following:

```jsonc
"cSpell.languageSettings": [
    {
        "locale": "en",
        "languageId": "*",
        "dictionaries": ["!en_us", "en-gb"]
    }
],
```

To turn off the US English dictionary and replace it with the British English dictionary.
